### ISSUE TITLE
Negative responses not being sent

### ISSUE THREAD

**Issue by Fanfwe** [avahi/avahi#428]
Hello,

I was reading through the RFC6762 and there's a paragraph saying:

> A Multicast DNS responder MUST only respond when it has a positive,
> non-null response to send, or it authoritatively knows that a
> particular record does not exist.  For unique records, where the host
> has already established sole ownership of the name, it MUST return
> negative answers to queries for records that it knows not to exist.
> For example, a host with no IPv6 address, that has claimed sole
> ownership of the name "host.local." for all rrtypes, MUST respond to
> AAAA queries for "host.local." by sending a negative answer
> indicating that no AAAA records exist for that name.

I have tried this on a linux debian host running avahi-daemon with the ipv6 stack disabled at kernel level.
Avahi was listening for mDNS requests, and I was sending mDNS AAAA queries over ipv4 for its <host>.local. address and it was not sending any negative response, just silently ignoring my requests. (Same query for <host>.local. as a type A was properly being responded as expected)

Is this expected behavior ? Is there a way to configure Avahi to send those negative responses ?

**Comment by evverx** [comment]
Currently NSEC resource records aren't supported. The reflector doesn't reflect them correctly either: https://github.com/lathiat/avahi/issues/379

**Comment by nkarstens** [comment]
Just want to chime in that this would be useful functionality.

**Comment by tgrushka** [comment]
It is now February 2026, over 3 years later.

You are active. You've patched CVEs and merged pull requests recently. @Fanfwe  explicitly called out the RFC requirement, @pemensik, a core developer on this project, acknowledged it's "what we are missing" and labeled it "important" two years ago. Yet three years later, no fix. The problem is that you treat RFC compliance as a feature request, not a bug. You ship other changes (like disabling wide‑area DNS for a CVE) but ignore this core protocol violation that causes real‑world 5‑second delays and a bad user experience for many. That's inexcusable neglect of standards and users. This is a core RFC violation, a BUG, NOT a feature request.

**Comment by evverx** [comment]
> The problem is that you treat RFC compliance as a feature request, not a bug

I've just added the bug label.

> That's inexcusable neglect of standards and users

Feel free to switch to mDNSResponder. It's much better in terms of following the RFCs.

**Comment by pemensik** [comment]
The code and automatic testing don't materialize out of nowhere. There is no one paid for this work. Yes, I want it fixed. But I have a long list of things to fix. Is there a PR with a support I have missed?

**Comment by evverx** [comment]
> The code and automatic testing don't materialize out of nowhere

I don't think it should be necessary to explain anything. Those comments triggered the "Jia Tan energy" filter (loosely based on https://securelist.com/xz-backdoor-story-part-2-social-engineering/112476/). That being said it looks like the usual entitlement so all I can say is that if there are specific demands or requirements with specific timelines it's usually better to talk to vendors you have contractual relationship with. It's a bug tracker of an open source project.

> Is there a PR with a support I have missed?

No there isn't and please don't add the "needs help" label because it's likely to trigger the usual slop PRs "fixing" the issues with that label.
